test/integration/modules/edgecore/eventbus/eventbus_test.go

```diff
@@ -0,0 +1,215 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"database/sql"
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/kubeedge/kubeedge/test/integration/utils/common"
+
+	MQTT "github.com/eclipse/paho.mqtt.golang"
+	_ "github.com/mattn/go-sqlite3"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+//DeviceUpdate device update
+type DeviceUpdate struct {
+	State string `json:"state,omitempty"`
+}
+
+//the struct of device
+type Device struct {
+	id          []uint8
+	name        string
+	description string
+	state       string
+	lastOnline  []uint8
+}
+
+//Function to access the edge_core DB and return the device state.
+func getDeviceStateFromDB(deviceId string) string {
+	var device Device
+
+	pwd, err := os.Getwd()
+	if err != nil {
+		common.Failf("Failed to get PWD: %v", err)
+		os.Exit(1)
+	}
+	destpath := filepath.Join(pwd, "../../edge.db")
+	db, err := sql.Open("sqlite3", destpath)
+	if err != nil {
+		common.Failf("Open Sqlite DB failed :%v", err)
+	}
+	defer db.Close()
+	rows, err := db.Query("SELECT * FROM device", 0)
+	if err != nil {
+		common.Failf("Query Sqlite DB failed: %v", err)
+	}
+	defer rows.Close()
+	for rows.Next() {
+		err = rows.Scan(&device.id, &device.name, &device.description, &device.state, &device.lastOnline)
+		if err != nil {
+			common.Failf("Failed to scan DB rows: %v", err)
+		}
+
+		if string(device.id) == deviceId {
+			break
+		}
+	}
+	return device.state
+}
+
+//Run Test cases
+var _ = Describe("Event BUS Testing", func() {
+	var Token_client Token
+	var ClientOpts *MQTT.ClientOptions
+	var Client MQTT.Client
+	Context("Publish on eventbus topics throgh MQTT external broker", func() {
+		BeforeEach(func() {
+			ClientOpts = HubclientInit(ctx.Cfg.MqttEndpoint, Clientid, "", "")
+			Client = MQTT.NewClient(ClientOpts)
+			if Token_client = Client.Connect(); Token_client.Wait() && Token_client.Error() != nil {
+				common.InfoV6("client.Connect() Error is %s", Token_client.Error())
+			}
+			Expect(Token_client.Error()).NotTo(HaveOccurred())
+		})
+		AfterEach(func() {
+			common.PrintTestcaseNameandStatus()
+		})
+
+		It("TC_TEST_EBUS_1 :Sending Messge on SYS/dis/upload_records Topic", func() {
+
+			if Token_client = Client.Publish(UploadRecordToCloud, 0, false, "messagetoUpload_record_to_cloud"); Token_client.Wait() && Token_client.Error() != nil {
+				common.InfoV6("client.Publish() Error is %s", Token_client.Error())
+			} else {
+				common.InfoV6("client.Publish Success !!")
+			}
+			Expect(Token_client.Error()).NotTo(HaveOccurred())
+
+			Client.Disconnect(1)
+		})
+
+		It("TC_TEST_EBUS_2 :Sending Messge on $hw/events/device/+/state/update Topic", func() {
```

Comment:
can we simplify the descriptions rather than mentioning the topic names in each test case description ? it will help somebody new to understand better.